lib/gui/components/svg-icon/svg-icon.js

```diff
@@ -21,8 +21,94 @@
  */
 
 const angular = require('angular');
+const react = require('react');
+const propTypes = require('prop-types');
+const react2angular = require('react2angular').react2angular;
+const path = require('path');
+const fs = require('fs');
+const _ = require('lodash');
+
 const MODULE_NAME = 'Etcher.Components.SVGIcon';
-const SVGIcon = angular.module(MODULE_NAME, []);
-SVGIcon.directive('svgIcon', require('./directives/svg-icon'));
+const angularSVGIcon = angular.module(MODULE_NAME, []);
+
+/**
+ * @summary SVG element that takes both filepaths and file contents
+ * @type {Object}
+ * @public
+ */
+class SVGIcon extends react.Component {
+
+  /**
+   * @summary Render the SVG
+   * @returns {react.Element}
+   */
+  render() {
+    let contents = null;
+
+    // The path contains SVG contents
+    if (_.first(this.props.path) === '<') {
+      contents = this.props.path;
+
+    } else {
+
+      // This means the path to the icon should be
+      // relative to *this directory*.
+      // TODO: There might be a way to compute the path
+      // relatively to the `index.html`.
+      const imagePath = path.join(__dirname, this.props.path);
```

Comment:
I don't think we should hardcore the path to the main HTML file from this directive, since this component shouldn't care about that. I agree that we can improve the situation, but looks like it needs to be done when passing the path to this component, rather than on this component itself.